src/ortho.rs

```diff
@@ -1,23 +1,162 @@
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct Ortho {
     version: u64,
+    storage: Vec<u16>,
+    dimensions: Vec<u16>,
 }
 
 impl Ortho {
     pub fn new(version: u64) -> Self {
-        Ortho { version }
+        // Use minimum dimensions [2,2] for orthogonality
+        Ortho { 
+            version,
+            storage: Vec::new(),
+            dimensions: vec![2, 2],
+        }
+    }
+    
+    pub fn with_dimensions(version: u64, dimensions: Vec<u16>) -> Self {
+        // Ensure minimum dimensions are at least [2,2]
+        let dims = if dimensions.len() < 2 || dimensions.iter().any(|&d| d < 2) {
+            vec![2, 2]
+        } else {
+            dimensions
+        };
+        
+        Ortho {
+            version,
+            storage: Vec::new(),
+            dimensions: dims,
+        }
     }
 
     pub fn version(&self) -> u64 {
         self.version
     }
+    
+    /// Generate all logical coordinates sorted by shell (sum) then by components
+    fn generate_logical_coordinates(&self) -> Vec<Vec<u16>> {
+        // Generate Cartesian product of all dimension ranges
+        let coords = self.cartesian_product(vec![], 0);
```

Comment:
Don't pass these args in to cartesian product. Don't have cartesian product on self. Just have it be a regular function that takes in dims and gives back the cartesian product of all vectors that result